ui/components/ImageAutomation/repositories/ImagePolicy.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { useGetObject } from "../../../hooks/objects";
+import { Kind } from "../../../lib/api/core/types.pb";
+import { FluxObject } from "../../../lib/objects";
+import Flex from "../../Flex";
+import InfoList from "../../InfoList";
+import PageStatus from "../../PageStatus";
+import RequestStateHandler from "../../RequestStateHandler";
+import Spacer from "../../Spacer";
+import Text from "../../Text";
+
+type Props = {
+  name: string;
+  namespace: string;
+  clusterName: string;
+};
+const ImagePolicy = ({ name, namespace, clusterName }: Props) => {
+  const { data, isLoading, error } = useGetObject<FluxObject>(
+    name,
+    namespace,
+    Kind.ImagePolicy,
+    clusterName,
+    {
+      refetchInterval: 5000,
+    }
+  );
+  return (
+    <RequestStateHandler loading={isLoading} error={error}>
+      {!!data && (
+        <Flex wide tall column>
+          <Text size="large" semiBold titleHeight>
+            Policy
+          </Text>
+          <Spacer margin="xs" />
+          <PageStatus conditions={data.conditions} suspended={data.suspended} />
+          <Spacer margin="xs" />
+          <InfoList
+            items={[
+              ["Image Policy", Object.keys(data.obj.spec.policy)[0]],
+              ["Order/Range", getValueByKey(data.obj.spec.policy, "range")],
```

Comment:
This is why I was wondering if ImagePolicy was meant to have it's own class - will we always be able to read obj.spec.policy or is there a chance that it will be undefined?